src/client/ua_client_highlevel_subscriptions.c

```diff
@@ -192,6 +192,7 @@ UA_Client_Subscriptions_addMonitoredEvent(UA_Client *client, const UA_UInt32 sub
     UA_LOG_DEBUG(client->config.logger, UA_LOGCATEGORY_CLIENT,
                  "Created a monitored item with client handle %u", client->monitoredItemHandles);
 
+//    UA_EventFilter_delete(evFilter);
```

Comment:
This looks like a double free of a String, or you are assigning a non-malloced string to a member of evString and then the _delete tries to delete the string from the stack.

I'd guess it's the selectClause and whereClauses which are initialized outside of the method. Probably you should set `selectClausesSize` and `whereClausesSize` to 0 (before every `UA_EventFilter_delete(evFilter)`) so that these are not deleted within that method. And these should then be deleted by the callee.

Just a small note: To post code blocks on Github, use three backticks. I edited your comment :)